vault/programs/vault/src/context.rs

```diff
@@ -0,0 +1,323 @@
+use crate::zeta_context::*;
+use crate::*;
+
+#[derive(Accounts)]
+#[instruction(vault_name: String, vault_lamports: u64, bumps: VaultBumps)]
+pub struct InitializeVault<'info> {
+    // vault Authority accounts
+    #[account(mut)]
+    pub vault_authority: Signer<'info>,
+    // vault Accounts
+    #[account(init,
+        seeds = [vault_name.as_bytes()],
+        bump = bumps.vault,
+        payer = vault_authority)]
+    pub vault: Box<Account<'info, Vault>>,
+    // This is the PDA that holds SOL to pay for the margin account
+    #[account(mut,
+        seeds = [vault_name.as_bytes(), b"payer"],
+        bump = bumps.vault_payer)]
+    pub vault_payer: AccountInfo<'info>,
+    // TODO Confirm USDC mint address on mainnet or leave open as an option for other stables
+    #[account(constraint = usdc_mint.decimals == PLATFORM_PRECISION as u8)]
+    pub usdc_mint: Box<Account<'info, Mint>>,
+    #[account(init,
+        mint::decimals = PLATFORM_PRECISION as u8,
+        mint::authority = vault_payer,
+        seeds = [vault_name.as_bytes(), b"redeemable_mint"],
+        bump = bumps.redeemable_mint,
+        payer = vault_authority)]
+    pub redeemable_mint: Box<Account<'info, Mint>>,
+    #[account(init,
+        token::mint = usdc_mint,
+        token::authority = vault_payer,
+        seeds = [vault_name.as_bytes(), b"vault_usdc"],
+        bump = bumps.vault_usdc,
+        payer = vault_authority)]
+    pub vault_usdc: Box<Account<'info, TokenAccount>>,
+    // Programs and Sysvars
+    pub system_program: Program<'info, System>,
+    pub token_program: Program<'info, Token>,
+    pub rent: Sysvar<'info, Rent>,
+}
+
+#[derive(Accounts)]
+pub struct InitUserRedeemable<'info> {
+    // User Accounts
+    #[account(mut)]
+    pub user_authority: Signer<'info>,
+    #[account(init,
+        token::mint = redeemable_mint,
+        token::authority = vault_payer,
+        seeds = [user_authority.key().as_ref(),
```

Comment:
feels like in general (to me) the ordering of seeds should be
constant ("user redeemable"), seed unique to the context (vault_name), seed unique to the user (user_key).

Reply:
Agree with that